packages/bibtex/init.lua

```diff
@@ -58,6 +68,56 @@ local parseBibtex = function (fn)
    return entries
 end
 
+--- Resolve the 'crossref' and 'xdata' fields on a bibliography entry.
+-- (Supplementing the entry with the attributes of the parent entry.)
+-- Once resolved recursively, the crossref and xdata fields are removed
+-- from the entry.
+-- So this is intended to be called at first use of the entry, and have no
+-- effect on subsequent uses: BibTeX does seem to mandate cross refererences
+-- to be defined before the entry that uses it, or even in the same bibliography
+-- file.
+-- Implementation note:
+-- We are not here to check the consistency of the BibTeX file, so there is
+-- no check that xdata refers only to @xdata entries
+-- Removing the crossref field implies we won't track its use and implicitely
+-- cite referenced entries in the bibliography over a certain threshold.
+-- @tparam table bib Bibliography
+-- @tparam string key Valid entry key
+local function crossrefAndXDataResolve (bib, key)
+   local entry = bib[key]
+   local refs
+
+   local xdata = entry.attributes.xdata
+   if xdata then
+      entry.attributes.xdata = nil
+      refs = xdata and pl.stringx.split(xdata, ",")
+   end
+
+   local crossref = entry.attributes.crossref
+   if crossref then
+      refs = refs or {}
+      table.insert(refs, crossref)
+      entry.attributes.crossref = nil
+   end
+
+   if not refs then
+      return
+   end
+   for _, ref in ipairs(refs) do
+      local parent = bib[ref]
+      if parent then
+         crossrefAndXDataResolve(bib, ref)
+         for k, v in pairs(parent.attributes) do
+            if not entry.attributes[k] and k ~= "crossref" and k ~= "xdata" then
```

Comment:
Not to safe: normally we resolved the parent, so the two extra "and"'s are not needed